Source/ACE/Entity/Player.cs

```diff
@@ -2345,27 +2345,27 @@ public void HandleActionUse(ObjectGuid usedItemId)
         {
             new ActionChain(this, () =>
             {
-                if (CurrentLandblock != null)
+                if (GetInventoryItem(usedItemId) != null)
                 {
-                    // Just forward our action to the appropriate user...
-                    ActionChain onUseChain = new ActionChain();
-                    CurrentLandblock.ChainOnObject(onUseChain, usedItemId, (WorldObject wo) =>
+                    WorldObject iwo = GetInventoryItem(usedItemId);
+                    ////iwo = iwo.GetObjectFromWeenieType(); // is this right? is this even needed?
+                    iwo.InternalOnUse(Session);
+                }
+                else
+                {
+                    if (CurrentLandblock != null)
                     {
-                        UsableObject uo = wo as UsableObject;
-                        Portal p = wo as Portal;
-
-                        // FIXME: OnCollide for portals -- portals should be usable?
-                        if (p != null)
-                        {
-                            p.OnCollide(Guid);
-                        }
-
-                        if (uo != null)
+                        // Just forward our action to the appropriate user...
+                        ActionChain onUseChain = new ActionChain();
+                        CurrentLandblock.ChainOnObject(onUseChain, usedItemId, (WorldObject wo) =>
                         {
-                            uo.OnUse(Guid);
-                        }
-                    });
-                    onUseChain.EnqueueChain();
+                            if (wo != null)
```

Comment:
I don't believe this null check is needed -- the landblock will just drop the chain if the object isn't found (we actually may need to give the landblock some way of nacking the chainer of failure in the future).